type-generator/src/main/java/io/mesosphere/types/dtr/Main.java

```diff
@@ -0,0 +1,156 @@
+import io.mesosphere.types.dtr.models.internal.types.*;
+import io.mesosphere.types.dtr.models.parser.RAMLParser;
+import io.mesosphere.types.dtr.repository.internal.Repository;
+import io.mesosphere.types.dtr.repository.internal.RepositoryView;
+import org.apache.commons.cli.*;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+
+    /**
+     * Entry point
+     * @param args
+     */
+    public static void main(String[] args) throws ParseException {
+        Options options = new Options();
+        CommandLineParser parser = new DefaultParser();
+
+        // Prepare options
+        Option o_help = Option.builder("h")
+                .longOpt("help")
+                .desc("Print this help message")
+                .argName("help")
+                .build();
+
+        Option o_project = Option.builder("p")
+                .longOpt("project")
+                .required()
+                .desc("The project name to target")
+                .hasArg()
+                .argName("project")
+                .build();
+
+        Option o_version = Option.builder("v")
+                .longOpt("version")
+                .required()
+                .desc("The project version to use")
+                .hasArg()
+                .argName("version")
+                .build();
+
+        Option o_repo = Option.builder("r")
+                .longOpt("repo")
+                .required()
+                .desc("The repository URI to use")
+                .hasArg()
+                .argName("repo")
+                .build();
+
+        // Compose options
+        options.addOption(o_help);
+        options.addOption(o_project);
+        options.addOption(o_version);
+        options.addOption(o_repo);
+
+        // Parse options
+        try {
+            CommandLine cmd = parser.parse(options, args);
+
+            // Handle and bail early on hep argument
+            if (cmd.hasOption("help")) {
+                HelpFormatter formatter = new HelpFormatter();
+                formatter.printHelp( "dtr", options );
+                return;
+            }
+
+            // Create a repository that is going to be our access
+            // channel to the type registry.
+            Repository repo = Repository.fromURI(cmd.getOptionValue("repo"));
+
+            // Access the specified project & version
+            RepositoryView project = repo.openProjectTypes(
+                cmd.getOptionValue("project"),
+                cmd.getOptionValue("version")
+            );
+
+            // Load project
+            RAMLParser raml = new RAMLParser();
+            GlobalScope globalScope = new GlobalScope();
+            raml.loadProjectTypes(project, globalScope);
+
+            // Initialize template
+            RepositoryView compilerView = repo.openProjectCompiler(
+                cmd.getOptionValue("project"),
```

Comment:
What is the difference between `openProjectTypes` and `openProjectCompiler`?

Reply:
Both of them return a view to different folders.

* `openProjectTypes` gives access to `/types` folder
* `openProjectCompiler ` gives access to the `/compiler` folder